#!/usr/bin/env python
# Created on: Oct 2, 2012
#
#       Author: Pablo Inigo Blasco - Robotics and Computer Architecture (ATC) - University of Seville - 2012
#       License: GPLv3
#http://www.ros.org/wiki/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile

PACKAGE = "rtcus_navigation"
import roslib;roslib.load_manifest(PACKAGE)
from dynamic_reconfigure.parameter_generator import *
gen = ParameterGenerator()

gen.add("planner_frequency", double_t, 0, "The frequency rate in Hz of the obstacle avoidance (navigation) loop", 50,0.1,1000.0)
gen.add("reference_frame", str_t, 0, "", "/odom")
gen.add("base_link_frame", str_t, 0, "", "/base_link")
gen.add("prediction_base_link_frame", str_t, 0, "", "/base_link_prediction")

size_enum = gen.enum([ gen.const("A", str_t, "rtcus_navigation/default_nonholonomic_DynamicState2D_Twist2D", "State prediction"),
                       gen.const("B", str_t, "rtcus_navigation/non_time_DynamicState2D_Twist2D", "Using the last state reading"),
                       gen.const("C", str_t, "state_estimation_type: rtcus_navigation/nonholonomic_trajectory_rollout2d", "Trayectory rollout considering Using the dynamics of the vehicle")  ],
                  "State estimation type")

gen.add("state_estimation_type", str_t, 0, "rtcus_navigation/default_nonholonomic_DynamicState2D_Twist2D",  edit_method=size_enum)

#gen.add("state_estimation_type", str_t, 0, "The type of the default state estimator module loaded by the pluginlib system.", "rtcus_navigation/non_time_DynamicState2D_Twist2D")
 
exit(gen.generate(PACKAGE, "navigation_node", "NavigationArchitecture"))

